### PR TITLE
[i18nIgnore] i18n(zh-cn): Fix Chinese subtag

### PR DESCRIPTION
**Pull request by liruifengv**
#### What kind of changes does this PR include?

- Minor content fixes (broken links, typos, etc.)
- Changes to Starlight code

#### Description

- Fix Chinese subtag
- Url: zh -> zh-cn
- Built-in Translations: zh -> zh-CN
- All link /zh/ -> /zh-cn/
- Add redirect: /zh/*    /zh-cn/:splat

### CODE REVIEW

**Comment by liruifengv** [comment]
related PR: https://github.com/withastro/starlight/pull/475
discuss: https://discord.com/channels/830184174198718474/1160994417470079138

**Comment by Genteure** [comment]
Some thoughts:

- This change would be better to make after [existing i18n PRs to docs](https://github.com/withastro/starlight/pulls?q=is%3Apr+is%3Aopen+sort%3Aupdated-desc+i18n+zh) are merged.
- Links can be mass search-replaced in the same PR (`/zh/` -> `/zh-cn/`).
- A changeset is needed as this would be a breaking change.

File: docs/public/_redirects [hunk 1]
```diff
@@ -1,2 +1,3 @@
 /ph/*        https://astro-houston-ph.pages.dev/ph/:splat  200
 /:lang/*    /:lang/404/   404
+/zh/*       /zh-cn/:splat  200
```
Comment (liruifengv) [inline docs/public/_redirects:3]: How to test the redirect?

Comment (Genteure) [reply docs/public/_redirects:3]: Redirect not working, getting HTTP 404 on this page: https://deploy-preview-855--astro-starlight.netlify.app/zh/getting-started/

Comment (liruifengv) [reply docs/public/_redirects]: I got it. It's order!

File: packages/starlight/translations/index.ts [hunk 2]
```diff
@@ -15,7 +15,7 @@ import da from './da.json';
 import tr from './tr.json';
 import ar from './ar.json';
 import nb from './nb.json';
-import zh from './zh.json';
+import zhCN from './zh-CN.json';
```
Comment (delucis) [inline packages/starlight/translations/index.ts:18]: As @Genteure suggested, we can probably stick with this being `zh` here. That will mean people already using `lang: 'zh'` can keep using this. And our system is able to pick this up anyway, so sites using `lang: 'zh-CN'` will also get this file.

```suggestion
import zh from './zh-CN.json';
```

File: packages/starlight/translations/index.ts [hunk 3]
```diff
@@ -41,7 +41,7 @@ export default Object.fromEntries(
 		tr,
 		ar,
 		nb,
-		zh,
+		"zh-CN": zhCN,
```
Comment (delucis) [inline packages/starlight/translations/index.ts:44]:
```suggestion
		zh,
```

**Review by delucis — COMMENTED** [review]
Thanks for updating this @liruifengv!

Just one comment on what we do with our built-in translation dictionary.

**Comment by Genteure** [comment]
@delucis Do you think the changeset is still needed, or it can be removed since this ended up mostly changing starlight's own docs? I suggested adding a changeset because I originally thought this would be a breaking change for existing users with Chinese docs in their project.

File: packages/starlight/translations/zh-CN.json
[the remark this answers is not in the document]
Comment (liruifengv) [reply packages/starlight/translations/zh-CN.json:1]: I think it's great, too.

**Comment by delucis** [comment]
Great point @Genteure. Yes, this is now a 100% transparent change that doesn’t impact Starlight users really, so we can remove the changeset.

**Comment by liruifengv** [comment]
> Great point @Genteure. Yes, this is now a 100% transparent change that doesn’t impact Starlight users really, so we can remove the changeset.

I removed it. It doesn’t impact existing users because we kept the built-in UI as zh

**Review by delucis — APPROVED** [review]
Great — I think this PR is ready to merge then. Thanks for all the work on this!

**Pull request merged**